darkmoon/cli/utils.py

```diff
@@ -105,25 +101,26 @@ def get_metadata(file: Path, source_iso: Path) -> dict[str, Any]:
     }
     if file_extension == ".exe" or file_extension == ".dll":
         try:
-            data_fields["header_info"] = get_all_exe_metadata(source_iso)
+            data_fields["header_info"] = get_all_exe_metadata(file)
         except PEFormatError:
             pass
-
+    else:
+        data_fields["header_info"] = "Not an EXE or DLL file"
     return data_fields
 
 
-def extract_files(file: Path, source_iso: Path, url: Path) -> None:
+def extract_files(file: Path, source_iso: Path, url: str) -> None:
     """Extract vmdk and put in new folder."""
     with tempfile.TemporaryDirectory() as tmpdirname:
-        with tarfile.open(file) as f:
-            f.extractall(tmpdirname)
+        cmd = ["7z", "x", str(file), "-o" + tmpdirname]
+        subprocess.run(cmd, check=True)  # noqa: S603
```

Comment:
Please move the S603 to the pyproject.toml as a globally excluded value, rather than just on this one line.